R/bm-array-altrep-materialization.R

```diff
@@ -0,0 +1,119 @@
+#' Benchmark for materializing an altrep Arrow array
+#'
+#' This flexes a lower level conversion to R data structures from Arrow data structures.
+#'
+#' @section Parameters:
+#' * `source` A known-file id to use (it will be read in to a data.frame first)
+#' * `exclude_nulls` logical, should any columns with any `NULL`s or `NA`s in them be removed?
+#'
+#' @importFrom purrr transpose
+#' @export
+array_altrep_materialization <- Benchmark(
+  "array_altrep_materialization",
+
+  setup = function(
+      source = names(known_sources),
+      exclude_nulls = FALSE,
+      altrep = TRUE
+  ) {
```

Comment:
Should we add stylr hooks to autostyle things like this? I suspect lintr will complain about _a lot_ of arrowbench because of the way it needs to do what it does, but styler should work ok + help make these easy.

Reply:
As long as it doesn't break anything I'm in. The code isn't in too bad of shape that way though (imo), so it doesn't seem urgent.